api/tests/src/api/agencies_v1/test_agencies_routes.py

```diff
@@ -111,3 +119,56 @@ def test_agencies_sorting(self, client, api_auth_token, enable_factory_create, d
         # assert order by agency_name desc then agency_code desc
         assert data[0]["agency_name"] > data[1]["agency_name"]
         assert data[1]["agency_code"] > data[2]["agency_code"]
+
+    def test_agencies_active(self, client, api_auth_token, enable_factory_create, db_session):
+        # Setup data
+        hhs = AgencyFactory.create(agency_name="HHS", agency_code="DOI")
+        crees = AgencyFactory.create(agency_name="CSREE", agency_code="USDA-CSREE")
+        darpa = AgencyFactory.create(agency_name="DARPA", agency_code="DOD-DARPA-MTO")
+
+        # HHS
+        OpportunityFactory.create(agency_code=hhs.agency_code)  # POSTED
+        OpportunityFactory.create(agency_code=hhs.agency_code)  # POSTED
+
+        # CREES
+        opp_2 = OpportunityFactory.create(
+            agency_code=crees.agency_code, current_opportunity_summary=None
+        )
+        opp_3 = OpportunityFactory.create(
+            agency_code=crees.agency_code, current_opportunity_summary=None
+        )
+        CurrentOpportunitySummaryFactory.create(
+            opportunity=opp_2, opportunity_status=OpportunityStatus.CLOSED
+        )
+        CurrentOpportunitySummaryFactory.create(
+            opportunity=opp_3, opportunity_status=OpportunityStatus.FORECASTED
+        )
+
+        # DARPA
+        opp_4 = OpportunityFactory.create(
+            agency_code=darpa.agency_code, current_opportunity_summary=None
+        )
+        CurrentOpportunitySummaryFactory.create(
+            opportunity=opp_4, opportunity_status=OpportunityStatus.ARCHIVED
+        )
```

Comment:
Something I setup to make this a lot easier, you don't need to create them separate, there's a param you can pass to the factory to make the opportunity summary a certain status:
```suggestion
        opp_4 = OpportunityFactory.create(
            agency_code=darpa.agency_code, is_archived_non_forecast_summary=True
        )
```
There is one for each of the statuses

Reply:
Nice! updated